providers/implementations/ciphers/cipher_aes_ccm_hw.c

```diff
@@ -29,6 +30,9 @@ static int ccm_generic_aes_initkey(PROV_CCM_CTX *ctx, const unsigned char *key,
 {
     PROV_AES_CCM_CTX *actx = (PROV_AES_CCM_CTX *)ctx;
 
+    if (!provider_is_running())
```

Comment:
This looks like overkill to me..
You should only need to override top level dispatch operations right?
Which is these..
```
OSSL_FUNC_CIPHER_NEWCTX, 
OSSL_FUNC_CIPHER_FREECTX
OSSL_FUNC_CIPHER_ENCRYPT_INIT,
OSSL_FUNC_CIPHER_DECRYPT_INIT,
OSSL_FUNC_CIPHER_UPDATE
OSSL_FUNC_CIPHER_FINAL
OSSL_FUNC_CIPHER_CIPHER
....
...
```

The initkey should be called from the init..

Reply:
I might have been overzealous.

In some cases the encrypt init and decrypt init call through to a common routine and it was better to put the check there.  I figured this was such a case.